java/src/main/java/de/gdata/vaas/Vaas.java

```diff
@@ -1,655 +1,278 @@
 package de.gdata.vaas;
 
 import de.gdata.vaas.messages.*;
+import de.gdata.vaas.options.ForFileOptions;
+import de.gdata.vaas.options.ForSha256Options;
+import de.gdata.vaas.options.ForStreamOptions;
+import de.gdata.vaas.options.ForUrlOptions;
 import de.gdata.vaas.exceptions.*;
 import lombok.Getter;
 import lombok.NonNull;
-import org.java_websocket.exceptions.WebsocketNotConnectedException;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URI;
-import java.net.URISyntaxException;
 import java.net.URL;
+import java.net.URLEncoder;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
-import java.net.http.HttpClient.Version;
 import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
 import java.security.NoSuchAlgorithmException;
+import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
 
-public class Vaas implements AutoCloseable {
-    private static final int connectionRetryDelayInMs = 1000;
-    private static final int connectionTimeoutInMs = 10000;
+public class Vaas implements IVaas {
+    private static final String userAgent = "Java/9.0.0";
```

Comment:
Get version at runtime.

Reply:
Java and Gradle does not support this, has to be implemented in a seperated logic